clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go

```diff
@@ -0,0 +1,233 @@
+// If namespace is not passed by parameter, all-namespace scope is assumed.
+func (*WaitForControlledPodsRunningMeasurement) Execute(config *measurement.MeasurementConfig) ([]measurement.Summary, error) {
+	var summaries []measurement.Summary
+	kind, err := util.GetString(config.Params, "kind")
+	if err != nil {
+		return summaries, err
+	}
+	namespace, err := util.GetStringOrDefault(config.Params, "namespace", metav1.NamespaceAll)
+	if err != nil {
+		return summaries, err
+	}
+	labelSelector, err := util.GetStringOrDefault(config.Params, "labelSelector", "")
+	if err != nil {
+		return summaries, err
+	}
+	fieldSelector, err := util.GetStringOrDefault(config.Params, "fieldSelector", "")
+	if err != nil {
+		return summaries, err
+	}
+	timeout, err := util.GetDurationOrDefault(config.Params, "timeout", defaultWaitForRCPodsTimeout)
+	if err != nil {
+		return summaries, err
+	}
+
+	runtimeObjectsList, err := getRuntimeObjectForKind(config.ClientSet, kind, namespace, labelSelector, fieldSelector)
+	if err != nil {
+		return summaries, err
+	}
+	runningRuntimeObjects := 0
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(defaultWaitForRCPodsInterval) {
+		if runningRuntimeObjects >= len(runtimeObjectsList) {
+			return summaries, nil
+		}
+
+		runtimeObjectNamespace, err := getNamespaceRuntimeObject(runtimeObjectsList[runningRuntimeObjects])
+		if err != nil {
+			return summaries, err
+		}
+		runtimeObjectSelector, err := getSelectorFromRuntimeObject(runtimeObjectsList[runningRuntimeObjects])
+		if err != nil {
+			return summaries, err
+		}
+		runtimeObjectReplicas, err := getReplicasFromRuntimeObject(runtimeObjectsList[runningRuntimeObjects])
+		if err != nil {
+			return summaries, err
+		}
+
+		err = waitForPods(config.ClientSet, runtimeObjectNamespace, runtimeObjectSelector.String(), "", int(runtimeObjectReplicas), timeout-time.Since(start))
+		if err != nil {
+			return summaries, fmt.Errorf("waiting for %s pods error: %v", kind, err)
+		}
+		runningRuntimeObjects++
+		if namespace == metav1.NamespaceAll {
+			glog.Infof("%s: %d %ss with all pods running", waitForControlledPodsRunningName, runningRuntimeObjects, kind)
+		} else {
+			glog.Infof("%s: %v: %d %ss with all pods running", waitForControlledPodsRunningName, namespace, runningRuntimeObjects, kind)
+		}
+	}
+
+	return summaries, fmt.Errorf("timeout while waiting for %ss to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running with all pods", kind, namespace, labelSelector, fieldSelector, runningRuntimeObjects)
+}
+
+func getRuntimeObjectForKind(c clientset.Interface, kind, namespace, labelSelector, fieldSelector string) ([]runtime.Object, error) {
+	listOpts := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	runtimeObjectsList := make([]runtime.Object, 0)
+	switch kind {
+	case "ReplicationController":
+		list, err := c.CoreV1().ReplicationControllers(namespace).List(listOpts)
```

Comment:
Why can't we use dynamic client as everywhere else?

Reply:
I guess we can. I just need to add dynamic client to measurement config.
However, I definitely don't want to this in this PR.

Reply:
Ping @wojtek-t. If you're ok with it.